app/views/products/index.html.haml

```diff
@@ -31,19 +33,38 @@
           %td
             = product.price
           %td
-            = link_to 'View product', product_path(product), class: 'btn btn-outline-primary me-3',
-            data: {turbolinks: false}
-            = link_to 'Edit product', edit_product_path(product), class: 'btn btn-outline-warning',
-            data: {turbolinks: false}
+            .d-flex
+              - if check_if_admin?
+                = link_to 'View product', product_path(product), class: 'btn btn-outline-primary me-3',
+                          data: { turbolinks: false }
+                = link_to 'Edit product', edit_product_path(product), class: 'btn btn-outline-warning',
+                          data: { turbolinks: false }
+              - else
+                = link_to 'View product', user_product_path(product), class: 'btn btn-outline-info me-2'
+                - if current_cart.cart_items.where(product_id: product.id).empty?
+                  = bootstrap_form_with model: @cart_item do |f|
+                    = f.hidden_field :product_id, value: product.id
+                    = f.hidden_field :quantity, value: 1
+                    = f.submit "Add To Cart", class: 'btn btn-outline-primary', id: "add_#{product.id}"
+                - else
+                  = bootstrap_form_with model: current_cart.cart_items.where(product_id: product.id).first do |f|
+                    = f.hidden_field :product_id, value: product.id
+                    = f.hidden_field :quantity,
+                      value: current_cart.cart_items.where(product_id: product.id).first.quantity + 1
+                    = f.submit "Add To Cart", class: 'btn btn-outline-primary', id: "add_#{product.id}"
```

Comment:
let's move this form into separate partial, and use find_by there instead of where...first